R/pkg/R/generics.R

```diff
@@ -439,6 +439,10 @@ setGeneric("covar_samp", function(col1, col2) {standardGeneric("covar_samp") })
 #' @export
 setGeneric("covar_pop", function(col1, col2) {standardGeneric("covar_pop") })
 
+#' @rdname dapply
+#' @export
+setGeneric("dapply", function(x, func, schema = NULL) { standardGeneric("dapply") })
```

Comment:
My vote would be to gradually add more complexity
1. `dapply` with schema - this PR
2. `dapplyCollect` needing no schema - a follow up PR
3. Say auto-schema detection in the worker.R file to figure out the schema of the resulting data.frame in the UDF. If the auto-schema detection fails, we fall back to serialized R data for specific columns etc. But this will be complex and can wait. 

@rxin @davies Does this sound good ?